### ISSUE TITLE
Resource hog and slow startup

### ISSUE THREAD

**Issue by tormodatt** [fwcd/kotlin-language-server#248]
The file-walk change made in this commit https://github.com/fwcd/kotlin-language-server/commit/adff2f93f43d6fd10635ad2a4d81226c4348c37b makes my fan go crazy and makes the server use several minues to start up. If I revert that commit it takes seconds, and no fan.

I'm testing on a small project on macOS with nvim-lsp. Have the same issue with coc.nvim on stable vim.

I might be able to look further on why, some other time.

**Comment by fwcd** [comment]
I think the issue is that Files.walk forces the language server to traverse the entire tree instead of cutting of entire branches when an ignored pattern is found.

**Comment by tormodatt** [comment]
Thank you for your fast response! However, I tested latest master, and I still have the same issue. If I do the reverts it starts up much faster on this repo.

I did some quick startup-time tests:

The repo I tested before: 3+ minutes

Git repo with only one file (same file as I open in the repo above): 2 secs

Another repo (bigger in size compared to the first one): 20-30 sec


So it seems that it still goes through more files than necessary.

I'll try to run the function from shared/src/main/kotlin/org/javacs/kt/classpath/DefaultClassPathResolver.kt on that path and print out the files it's going through to determine which files that might be later.
